### DISCUSSION TITLE
Potential bug about named nodes

### DISCUSSION THREAD

**Discussion by Lotes** [eclipse-langium/langium#486] [General]
Looking at these lines:
https://github.com/langium/langium/blob/edf9bb1b85dd8ea20cf0f4043547d305779f5df2/packages/langium/src/references/naming.ts#L42-L44

Is it possible that the `getNameNode(...)` call returns a node whose name is not string and is this a problem? If we have an example use case, we may have a bug.

**Comment by spoenemann** [comment]
Right. The easiest fix might be this:
```
if (typeof (node as NamedAstNode).name === 'string') {
    return findNodeForFeature(node.$cstNode, 'name'); 
}
```
